### ISSUE TITLE
require-returns not working for async functions that don't return anything

### ISSUE THREAD

**Issue by AndrewLeedham** [gajus/eslint-plugin-jsdoc#276]
Seems like https://github.com/gajus/eslint-plugin-jsdoc/issues/174 has resurfaced in later versions. I created a minimal repro that has the same issue when running with both TS and JS, just run `yarn lint:ts` and `yarn lint:js` to see the issue: https://github.com/AndrewLeedham/eslint-plugin-jsdoc-repro

**ESLint version**: 5.16.0
**eslint-plugin-jsdoc version**: 8.0.1
**Node version**: v10.15.3

**Comment by gajus** [comment]
CC @golopot

**Comment by golopot** [comment]
#174 address the rule `require-returns-check` while this issue addresses `require-returns`, so this is a new issue, not a regression.

I agree with you this is a false positive, however there are test cases saying your code is considered warning. So this issue might needs decision from the maintainers. @gajus @brettz9

https://github.com/gajus/eslint-plugin-jsdoc/blob/a9490947bb85960fbe41dae968a5875ff2085c0d/test/rules/assertions/requireReturns.js#L88-L104

I consider this a false positive:
```js
// Error: Missing JSDoc @returns declaration.
/**
 *
 */
async function quux() {}
```

**Comment by brettz9** [comment]
I personally like to use the force return setting, and even use the new settings to reject a plain child `Promise` (requiring it be of the parent form, i.e., `Promise<...>`, instead).

But I think others closer to my preferences may, even if they don't want reminders for plain `undefined` returns, still wish to be reminded that a Promise is being returned there, even if it is admittedly redundant with the `async`. I think some config to allow either would be desirable, though I'd think we could loosen the default to avoid insisting on a return for a mere async.

**Comment by AndrewLeedham** [comment]
So default would not flag an empty async function, but setting this new option to true or setting `forceRequireReturn` to true would flag it?

What would this new option be called `forcePromiseReturn` perhaps?

**Comment by brettz9** [comment]
Right, that was my thinking... But maybe it should be`forceAsyncReturn`, or more clearly, `forceReturnsWithAsync` to avoid giving the impression it has anything to do with normal `return Promise(...)`.

**Comment by AndrewLeedham** [comment]
Would it make sense to add the same setting to `require-returns-check`. As the test assertion
```
          /**
           * @returns {Promise<void>}
           */
          const quux = async () => {}
```
would fail, when defaulting an async function to not return an empty Promise.
